core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -235,6 +236,18 @@ private void enrichTopicOverview(
       topicInfo.setNoOfReplicas(topic.getNoOfReplicas());
       topicInfo.setTeamname(manageDatabase.getTeamNameFromTeamId(tenantId, topic.getTeamId()));
       topicInfo.setTeamId(topic.getTeamId());
+      String topicJsonParams = topic.getJsonParams();
+      if (topicJsonParams != null) {
+        TopicConfigurationRequest topicConfigurationRequest;
+        try {
+          topicConfigurationRequest =
+              OBJECT_MAPPER.readValue(topicJsonParams, TopicConfigurationRequest.class);
+          topicInfo.setAdvancedTopicConfiguration(
+              topicConfigurationRequest.getAdvancedTopicConfiguration());
+        } catch (JsonProcessingException e) {
+          throw new RuntimeException(e);
```

Comment:
Might be worth creating a KlawException that is caught in the controller advice that returns a bad request with a message per processable advanced config.